x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,379 @@
+                ...serializeTimeRange(),
+                ...serializeFieldStatsChartState(),
+              },
+              references,
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...fieldStatsControlsComparators,
+        }
+      );
+
+      const reload$ = fetch$(api).pipe(
+        skipWhile((fetchContext) => !fetchContext.isReload),
+        map(() => Date.now())
+      );
+      const reset$ = resetData$.pipe(skip(1), distinctUntilChanged());
+
+      const onTableUpdate = (changes: Partial<DataVisualizerTableState>) => {
+        if (isDefined(changes?.showDistributions)) {
+          fieldStatsControlsApi.showDistributions$.next(changes.showDistributions);
+        }
+      };
+
+      const addFilters = (filters: Filter[], actionId: string = ACTION_GLOBAL_APPLY_FILTER) => {
+        if (!pluginStart.uiActions) {
+          onError(new Error(ERROR_MSG.APPLY_FILTER_ERR));
+          return;
+        }
+        const trigger = pluginStart.uiActions.getTrigger(APPLY_FILTER_TRIGGER);
+        if (!trigger) {
+          onError(new Error(ERROR_MSG.APPLY_FILTER_ERR));
+          // eslint-disable-next-line no-console
+          console.error(`${ERROR_MSG.APPLY_FILTER_ERR}: APPLY_FILTER_TRIGGER is undefined`);
+        }
+        const actionContext = {
+          embeddable: api,
+          trigger,
+        } as ActionExecutionContext;
+
+        const executeContext = {
+          ...actionContext,
+          filters,
+        };
+        try {
+          const action = pluginStart.uiActions.getAction(actionId);
+          action.execute(executeContext);
+        } catch (error) {
+          onError(error);
+        }
+      };
+
+      const statsTableCss = css({
+        width: '100%',
+        height: '100%',
+        overflowY: 'auto',
+      });
+
+      return {
+        api,
+        Component: () => {
+          if (!apiHasExecutionContext(parentApi)) {
+            onError(new Error('Parent API does not have execution context'));
+          }
+
+          const { filters: globalFilters, query: globalQuery, timeRange } = useFetchContext(api);
+          const [dataViews, esqlQuery, viewType, showPreviewByDefault] =
+            useBatchedPublishingSubjects(
+              api.dataViews,
+              api.query$,
+              api.viewType$,
+              api.showDistributions$
+            );
+          const lastReloadRequestTime = useObservable(reload$, Date.now());
+
+          const isEsqlMode = viewType === FieldStatsInitializerViewType.ESQL;
+
+          const dataView =
+            Array.isArray(dataViews) && dataViews.length > 0 ? dataViews[0] : undefined;
+          const onAddFilter = (
+            field: string | DataViewField,
+            value: string,
+            operator: '+' | '-'
+          ) => {
+            if (!dataView || !pluginStart.data) {
+              onError(new Error(ERROR_MSG.APPLY_FILTER_ERR));
```

Comment:
Same comment about only using blocking errors when embeddable is broken in an unrecoverable way. If adding filters fails, the embeddable can still function. Instead of setting a blocking error, how about displaying a warning toast when addFilters fails?

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/0a39eb7364e235191c4f416dc3881afa8ef510e0